### ISSUE TITLE
False positive in type annotation of recursive data structure.

### ISSUE THREAD

**Issue by jonathanslenders** [PyCQA/pyflakes#567]
```python
from typing import Dict

def test() -> None:
    class Tree(Dict[str, "Tree"]):
        def __missing__(self, key: str) -> "Tree":
            self[key] = Tree()
            return self[key]
```
```
/tmp/x.py:4:26 undefined name 'Tree'
```

**Comment by asottile** [comment]
this as inadvertently fixed by #754

**Comment by jonathanslenders** [comment]
Thank you!
